pkg/frontend/changefeed.go

```diff
@@ -75,3 +90,47 @@ func (f *frontend) updateOcpVersions(docs []*api.OpenShiftVersionDocument) {
 		}
 	}
 }
+
+func (f *frontend) updateFromIteratorRoleSets(ctx context.Context, ticker *time.Ticker, frontendIterator cosmosdb.PlatformWorkloadIdentityRoleSetDocumentIterator) {
```

Comment:
Is there an opportunity to re-use the existing `updateFromIterator()` function?

Reply:
I don't think so, at least not without a non-trivial refactor.

The issue is that the generated database clients, specifically the ones for `PlatformWorkloadIdentityRoleSets` and `OpenShiftVersions` in this case, don't implement a common interface that we could leverage to deduplicate this code. I think it would require a refactor of the database client generation to add such an interface, and after doing that we would have to regenerate the clients and rewrite this code to use the interface.